src/registers.sw

```diff
@@ -0,0 +1,115 @@
+//! Functions to expose the 16 reserved FuelVM registers for ease of use.
+//! Ref: https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/main.md#semantics
+library registers;
+
+/// Contains zero (0), for convenience.
+pub fn zero() -> u64 {
+    asm() {
+        zero
+    }
+}
+
+/// Contains one (1), for convenience.
+pub fn one() -> u64 {
+    asm() {
+        one
+    }
+}
+
+/// Contains overflow/underflow of addition, subtraction, and multiplication.
+pub fn overflow() -> u64 {
+    asm() {
+        of
+    }
+}
+
+/// The program counter. Memory address of the current instruction.
+pub fn program_counter() -> u64 {
+    asm() {
+        pc
+    }
+}
+
+/// Memory address of bottom of current writable stack area.
+pub fn stack_start_ptr() -> u64 {
+    asm() {
+        ssp
+    }
+}
+
+/// Memory address on top of current writable stack area (points to free memory).
+pub fn stack_ptr() -> u64 {
+    asm() {
+        sp
+    }
+}
+
+/// Memory address of beginning of current call frame.
+pub fn frame_ptr() -> u64 {
+    asm() {
+        fp
+    }
+}
+
+/// Memory address below the current bottom of the heap (points to free memory).
+pub fn heap_ptr() -> u64 {
+    asm() {
+        hp
+    }
+}
+
+/// Error codes for particular operations.
+pub fn error() -> u64 {
+    asm() {
+        err
+    }
+}
+
+/// Remaining gas globally.
+pub fn global_gas() -> u64 {
+    asm() {
+        ggas
+    }
+}
+
+/// Remaining gas in the context.
+pub fn context_gas() -> u64 {
+    asm() {
+        cgas
+    }
+}
+
+/// Received balance for this context.
+pub fn balance() -> u64 {
+    asm() {
+        bal
+    }
+}
+
+/// Pointer to the start of the currently-executing code.
+pub fn instrs_start() -> u64 {
+    asm() {
+        is
+    }
+}
+
+/// Return value or pointer.
+// pub fn return_value() -> u64 {
+//     asm() {
+//         ret
+//     }
+// }
+
+/// Return value length in bytes.
+// pub fn return_length() -> u64 {
+//     asm() {
+//         retl
+//     }
+// }
```

Comment:
Is there a reason these are commented out?

Reply:
I'm investigating. It's because of this error:
```
error
  --> sway-lib-std/src/registers.sw:86:9
   |
84 |
85 |     asm() {
86 |         ret
   |         ^^^ This register was not initialized in the initialization section of the ASM expression. Initialized registers are:
87 |     }
88 | }
   |
error
  --> sway-lib-std/src/registers.sw:93:9
   |
91 |
92 |     asm() {
93 |         retl
   |         ^^^^ This register was not initialized in the initialization section of the ASM expression. Initialized registers are:
94 |     }
95 | }
   |
  Aborting due to 2 errors.
Error: "Failed to compile registers"
```

Reply:
It seems that `ret` and `retl` are not recognized as named registers.

Reply:
Looks like we're expecting a different spelling in the compiler: https://github.com/FuelLabs/sway/blob/916123ecda8a8a222fad63098ab36583871b2a13/sway-core/src/asm_lang/mod.rs#L1069
Should be an easy fix.

Reply:
oh yeah. 
Do we stick to the spec and rename in the compiler, or edit the spec and change them here?

Reply:
I think the spec is the reference we should follow. I can create a quick PR to update the compiler.